packages/core/src/extension/extension-service.ts

```diff
@@ -0,0 +1,58 @@
+import { noop } from 'lodash';
+
+import { IframeEventListener, IframeEventPoster } from '../common/iframe';
+
+import {
+    ExtensionCommandType,
+    ExtensionListenEventMap,
+    ExtensionListenEventType,
+    ExtensionPostCommand,
+} from './extension-client';
+
+export default class ExtensionService {
+    private _extensionId?: string;
+
+    constructor(
+        private _eventListener: IframeEventListener<ExtensionListenEventMap>,
+        private _eventPoster: IframeEventPoster<ExtensionPostCommand>,
+    ) {
+        this._eventPoster.setTarget(window.parent);
+    }
+
+    initialize(extensionId: string): void {
+        if (!extensionId) {
+            throw new Error('Extension Id not found.');
+        }
+
+        this._extensionId = extensionId;
+
+        this._eventListener.listen();
+    }
+
+    post(event: ExtensionPostCommand): void {
```

Comment:
```suggestion
    post(event: ExtensionCommand): void {
```
Because the client side only sends command, `post` in the name seems redundant.